### ISSUE TITLE
linter: `unicorn/no-array-callback-reference` and `unicorn/no-array-for-each` false positives on Effect library methods

### ISSUE THREAD

**Issue by eyenalxai** [oxc-project/oxc#18876]
Oxlint's unicorn rules incorrectly flag Effect library methods that share names with array methods, even though they operate on completely different types.

```ts
import { Effect, Console, Option } from "effect";

// eslint-plugin-unicorn(no-array-for-each): Do not use `Array#forEach`
const _result = Effect.forEach([1, 2, 3, 4, 5], (n, index) =>
  Console.log(`Currently at index ${index}`).pipe(Effect.as(n * 2)),
);

const oof = "s";

// eslint-plugin-unicorn(no-array-callback-reference): Avoid passing a function reference directly to iterator methods
const _someOof = Option.some(oof);
```

[Playground](https://playground.oxc.rs/?code=import+%7B+Effect%2C+Console+%7D+from+%22effect%22%0A%0Aconst+result+%3D+Effect.forEach%28%5B1%2C+2%2C+3%2C+4%2C+5%5D%2C+%28n%2C+index%29+%3D%3E%0A++Console.log%28%60Currently+at+index+%24%7Bindex%7D%60%29.pipe%28Effect.as%28n+*+2%29%29%0A%29%0A%0Aconst+oof+%3D+%22s%22%0A%0Aconst+someOof+%3D+Effect.some%28oof%29)
